apps/ledger-live-mobile/src/newArch/features/WalletSync/hooks/useAddMember.ts

```diff
@@ -0,0 +1,83 @@
+import { memberCredentialsSelector, setTrustchain } from "@ledgerhq/trustchain/store";
+import { useDispatch, useSelector } from "react-redux";
+import { useTrustchainSdk, runWithDevice } from "./useTrustchainSdk";
+import {
+  MemberCredentials,
+  TrustchainResult,
+  TrustchainResultType,
+} from "@ledgerhq/trustchain/types";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { Device } from "@ledgerhq/live-common/hw/actions/types";
+import { trace, listen } from "@ledgerhq/logs";
+
+listen(log => {
+  console.log(log);
+});
+
+export function useAddMember({ device }: { device: Device | null }) {
+  const dispatch = useDispatch();
+  const sdk = useTrustchainSdk();
+  const memberCredentials = useSelector(memberCredentialsSelector);
+  const [error, setError] = useState<Error | null>(null);
+
+  const [userDeviceInteraction, setUserDeviceInteraction] = useState(false);
+
+  const sdkRef = useRef(sdk);
+  const deviceRef = useRef(device);
```

Comment:
is this expected that deviceRef is cached forever?
i suspect you want to add a 
```
useEffect(()=> {
  deviceRef.current = device;
}, [device])
```

if this is intended, i suggest to rename this to `initialDeviceRef` because it's not getting updated